src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,512 @@
+                min={MONTHLY_MIN}
+                disabled={!(frequency === 'monthly')}
+                id="dayOfMonth"
+                name="dayOfMonth"
+                defaultValue={config ? config.dayOfMonth : 1}
+                onBlur={e => this.setState({
+                  invalidFields: {
+                    ...invalidFields,
+                    dayOfMonth: !e.target.value.length,
+                  },
+                })}
+              />
+            </ValidationFormGroup>
+          </div>
+          <div className="col">
+            <ValidationFormGroup
+              for="dayOfWeek"
+              helpText="The day of the week to send the report. This field is required and only valid when the frequency is weekly"
+            >
+              <label htmlFor="dayOfWeek">Day of Week</label>
+              <Input
+                type="select"
+                id="dayOfWeek"
+                name="dayOfWeek"
+                disabled={!(frequency === 'weekly')}
+                options={[
+                  { value: 0, label: 'Monday' },
+                  { value: 1, label: 'Tuesday' },
+                  { value: 2, label: 'Wednesday' },
+                  { value: 3, label: 'Thursday' },
+                  { value: 4, label: 'Friday' },
+                  { value: 5, label: 'Saturday' },
+                  { value: 6, label: 'Sunday' },
+                ]}
+                defaultValue={config ? config.dayOfWeek : undefined}
+              />
+            </ValidationFormGroup>
+          </div>
+          <div className="col">
+            <ValidationFormGroup
+              for="hourOfDay"
+              helpText="The hour of the day to send the report, in Eastern Standard Time (EST). This is required for all frequency settings"
+              invalid={invalidFields.hourOfDay}
+              invalidMessage="Required for all frequency types"
+            >
+              <label htmlFor="hourOfDay">Hour of Day</label>
+              <Input
+                type="number"
+                id="hourOfDay"
+                name="hourOfDay"
+                defaultValue={config ? config.hourOfDay : undefined}
+                onBlur={e => this.setState({
+                  invalidFields: {
+                    ...invalidFields,
+                    hourOfDay: !e.target.value.length,
+                  },
+                })}
+              />
+            </ValidationFormGroup>
+          </div>
+        </div>
+        <ValidationFormGroup
+          for="pgpEncyptionKey"
+          helpText="The key for encyption, if PGP encrypted file is required"
+        >
+          <label htmlFor="pgpEncryptionKey">PGP Ecryption Key</label>
+          <Input
+            type="textarea"
+            id="pgpEncryptionKey"
+            name="pgpEncryptionKey"
+            defaultValue={config ? config.pgpEncryptionKey : undefined}
+          />
+        </ValidationFormGroup>
+        {deliveryMethod === 'email' && (
```

Comment:
This component could probably be split into a few subcomponents. One good opportunity could be the conditional rendering based on `deliveryMethod`. For example:
```
{deliveryMethod === 'email' && <EmailDeliveryMethodFields />}
{deliveryMethod === 'sftp' && <SftpDeliveryMethodFields />}
```
There are other opportunities for subcomponents as well. This could help reduce the complexity and improve the readability of the `render` function.

Reply:
I've split out these two separate components. The fields are going to change slightly soon, so I haven't yet split it further, but if you think there's more to split out I'm open to suggestions.